### PR TITLE
LPS-119674 Cannot publish Staging over IPV6 connection

### PR DESCRIPTION
**Pull request by markocikos**
FIxes: https://issues.liferay.com/browse/LPS-119674

See details and steps to reproduce in the JIRA ticket. Note that it doesn't have to be the modal in staging. The bug is on all legacy modals, for example the one on Calendar widget in the gifs below.

@jbalsas correctly spotted the root cause of the issue in [this comment](https://issues.liferay.com/browse/LPS-119674?focusedCommentId=2499415&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2499415): the [regex](https://github.com/liferay/alloy-ui/blob/2c240fab9760bb33909d531c0f9753a5e4de3bd4/src/aui-url/js/aui-url.js#L38-L41) in AUI `Url` library only covers IPv4 format. In this PR, we are removing all four usages of `A.Url` in DXP, replacing them with JS native `URL`. It correctly covers IPv6 format.

Notes:
- I could not find use cases to live test usages outside modals, but the changes should be safe. I tested them all with sample URLs.
- I noticed in several places backend warnings for URLs:
```
... WARN  [PortalImpl:1044] Redirect URL http://[::1]:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_portal_workflow_kaleo_forms_web_portlet_KaleoFormsAdminPortlet&p_p_lifecycle=0&p_p_state=normal&p_p_state_rcv=1 is not allowed
```

Before PR:
before

After PR:
after

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/search_container_select.js [hunk 1]
```diff
@@ -206,9 +206,9 @@ AUI.add(
 				},
 
 				_isActionUrl(url) {
-					var uri = new A.Url(url);
+					const uri = new URL(url);
 
-					return Number(uri.getParameter('p_p_lifecycle')) === 1;
+					return Number(uri.searchParams.get('p_p_lifecycle')) === 1;
```
Comment (diegonvs) [inline modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/search_container_select.js:211]: Just a consideration here, if this code would be backported We must provide a polyfill for searchParams methods in older versions or use another solution. See [searchParams](https://caniuse.com/mdn-api_url_searchparams) browser support.

File: modules/apps/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/form_navigator_steps/steps.jspf [hunk 2]
```diff
@@ -255,13 +255,13 @@
 		var redirect = A.one('#<portlet:namespace />redirect');
 
 		if (redirect) {
-			var url = new A.Url(
+			const url = new URL(
 				redirect.val() ||
 					'<%= formNavigatorStepsDisplayContext.getPortletURL() %>'
 			);
 
-			url.setAnchor(null);
```
Comment (diegonvs) [inline modules/apps/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/form_navigator_steps/steps.jspf:263]: Same consideration as  https://github.com/liferay-frontend/liferay-portal/pull/1457/files#r706353122 here.

See https://caniuse.com/mdn-api_url_hash

**Review by diegonvs — APPROVED** [review]
LGTM :)

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/search_container_select.js (hunk 1 above)
Comment (jbalsas) [reply modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/search_container_select.js:211]: Remember we already have the [compatibility-ie](https://github.com/liferay/liferay-portal/tree/7.3.x/modules/apps/frontend-compatibility/frontend-compatibility-ie) module in `7.3.x` which [includes among others](https://github.com/liferay/liferay-portal/blob/7.3.x/modules/apps/frontend-compatibility/frontend-compatibility-ie/src/main/java/com/liferay/frontend/compatibility/ie/internal/servlet/taglib/IETopHeadDynamicInclude.java#L97-L99), the [core-js](https://github.com/zloirock/core-js/blob/master/README.md) bundle

**Comment by markocikos** [comment]
All green and compatibility concerns addressed. Pushing!

**Comment by markocikos** [comment]
ci:forward

**Pull request closed without merging**